client/components/newdeck.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+export default class NewDeck extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = ({ deckName: '' });
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    const { value } = event.target;
+    this.setState({ deckName: value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const req = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(this.state)
+    };
+    fetch('/api/create-deck', req)
+      .then(res => {
+        res.json();
+        this.props.closeModal();
+      })
+      .catch(err => console.error(err));
+    this.setState({ deckName: '' });
```

Comment:
I recommend delaying resetting state until your fetch completes. You can do that by moving this setState into the .then() callback.

Reply:
Change has been made here. I think this was the first one I wrote so I missed it when I had made the changes to waiting until fetch completes two features ago and just make it happen in the current feature (changing location)

![image](https://user-images.githubusercontent.com/109567499/201175259-1cf36f64-f0bc-4dbe-87fc-4cc0cd6bb7c7.png)